Upgrade webrtc lib in licode

j4y opened the pull request.

j4y committed:
update webrtc C++ classes to hopefully improve video quality

This was taken from
https://chromium.googlesource.com/external/webrtc

j4y committed:
add 'set -e' to the beginning of the install erizo bash script

This makes it easier to debug failed builds because the bash script will stop and show an error.

j4y committed:
add third_party/ to the gitignore file

This directory is used by the build and should not be in source control

j4y committed:
update karma loglevel to debug to see output

MorikoHandford commented on erizo/src/erizo/WebRtcConnection.cpp:
```diff
@@ -580,7 +583,8 @@ namespace erizo {
 //      p_.type = (transport->mediaType == VIDEO_TYPE) ? VIDEO_PACKET : AUDIO_PACKET;
       p_.type = type;
-      p_.length = length;
+      // TODO: modify dataPacket to use size_t
```
When is this change to modify dataPacket going to be done?

MorikoHandford commented on erizo/src/erizo/WebRtcConnection.h:
```diff
@@ -11,7 +11,9 @@
 #include "Transport.h"
 #include "Stats.h"
-#include "rtp/webrtc/fec_receiver_impl.h"
+// #include "rtp/webrtc/fec_receiver_impl.h"
```
Can this line just be removed?  We have it in the git history.

MorikoHandford commented on erizo/src/erizo/media/ExternalOutput.h:
```diff
@@ -3,7 +3,9 @@
 #include "../MediaDefinitions.h"
 #include "rtp/RtpPacketQueue.h"
-#include "rtp/webrtc/fec_receiver_impl.h"
+//#include "rtp/webrtc/fec_receiver_impl.h"
```
ditto removing this (and any further commented code).

MorikoHandford commented:
Except for the commented code, looks good to me.  :shipit:

j4y committed:
Merge branch 'master' into upgrade-webrtc-lib-merged

j4y committed:
update travis to use the container-based infrastructure and upgrade gcc/cmake/libboost

This speeds up travis builds and does not use sudo.

Gcc upgraded to 4.9.x, cmake upgraded to 3.2.3, libboost upgraded to 1.55

j4y committed:
remove the hard coded CMAKE_CXX_FLAG of '-std=c++0x' and use CheckCXXCompilerFlag to determine which flag is appropriate

This determines if the compiler supports C++11 and if it does adds in the '-std=c++11' flag
See http://stackoverflow.com/questions/10851247/how-to-activate-c-11-in-cmake for more information

j4y committed:
remove calls to octet_string_hex_string to unblock the upgrade to libsrtp 1.5.2

octet_string_hex_string is used in two log statements and doesn't appear to be available in libsrtp anymore:
https://github.com/cisco/libsrtp/blob/7d67ce5cd79c09d4de6dbe53b319c6bfed509e66/crypto/include/datatypes.h#L119

My attempts to include this datatypes.h into SrtpChannel failed.

j4y committed:
remove commented include to rtp/webrtc/fec_receiver_impl.h

j4y committed:
remove commented include to rtp/webrtc/fec_receiver_impl.h

j4y committed:
remove TODO about modifying dataPacket and static_cast

j4y committed:
remove cruft from MediaProcessor.cpp

j4y commented:
I'll send a new PR with the updated changes.
